### PR TITLE
Use ATL smart pointers for automatic memory management

### PR DESCRIPTION
**Pull request by lalitshankarch**
ATL smart pointers make the code simpler and thus there is no need to manually free resources.

### CODE REVIEW

File: drivers/wasapi/audio_driver_wasapi.h [hunk 1]
```diff
@@ -38,15 +38,16 @@
 #include "servers/audio_server.h"
 
 #include <audioclient.h>
+#include <atlbase.h>
```
Comment (bruvzg) [inline drivers/wasapi/audio_driver_wasapi.h:41]: Is it available for MinGW? I do not see it in my installation (MinGW 9.0.0).

Comment (lalitshankarchowdhury) [reply drivers/wasapi/audio_driver_wasapi.h:41]: Unfortunately no, ATL is a part of Visual C++.

Comment (bruvzg) [reply drivers/wasapi/audio_driver_wasapi.h:41]: If it's MSVC only, we can't use it in Godot. We want cross-compilation support, and use MinGW for all official releases.

Comment (lalitshankarchowdhury) [reply drivers/wasapi/audio_driver_wasapi.h:41]: Okay, no problem. 🙂

**Pull request closed without merging**